### PR TITLE
Support WebSocket protocol

### PR DESCRIPTION
**Pull request by nkuba**
Here we add support for WebSocket protocol. With this change the client will support all connection types defined by the electrum protocol: `tcp`, `ssl`, `ws` and `wss`.

The WebSocket client initialization expects remote server details to be provided as URL including a scheme and host, e.g. `tcp://electrum.io:50001`.

Examples of client initialization:
```go
// TCP
client, err := NewClient(ctx, "tcp://electrum.io:50001", nil)
// SSL
client, err := NewClient(ctx, "ssl://electrum.io:50002", nil)
// WS
client, err := NewClient(ctx, "ws://electrum.io:50003", nil)
// WSS
client, err := NewClient(ctx, "wss://electrum.io:50004", nil)
```

### CODE REVIEW

**Comment by nkuba** [comment]
Sample integration tests can be found in https://github.com/keep-network/keep-core/pull/3582.

File: electrum/transport_ws.go
Comment (lukasz-zimnoch) [inline electrum/transport_ws.go:111]: Looks like the comment is a bit misleading as we neither wait for the server response nor use a timeout.  Looks like this is taken from https://github.com/gorilla/websocket/blob/76ecc29eff79f0cedf70c530605e486fc32131d1/examples/echo/client.go#L71 where the above is actually done.

According to WS specification (https://datatracker.ietf.org/doc/html/rfc6455#section-7.1.1) the flow for closing the connection should be:
1. Client asks for closing the connection
2. Server closes the connection and notifies the client
3. The client waits for closure notification. If not received after a reasonable amount of time, the client may close the connection on their own

I think we should implement the above pattern here.

File: electrum/transport_ws.go
Comment (lukasz-zimnoch) [inline electrum/transport_ws.go:29]: Should we enforce that `tlsConfig` must be `nil` for `ws` connections or that's fine to not bother about it at all? Conversely, shouldn't we enforce it is non-nil for `wss`?

File: electrum/transport_ws.go
Comment (nkuba) [reply electrum/transport_ws.go:111]: d5f2ae327a241e525c5e300a85ab364c8324ce67

File: electrum/transport_ws.go
Comment (nkuba) [reply electrum/transport_ws.go:29]: I don't think we should enforce `tlsConfig` presence or absence. It's optional and the application using the client should decide if it needs to provide any additional config.

**Review by lukasz-zimnoch — APPROVED** [review]

**Pull request merged**
